src/pip/_internal/pep425tags.py

```diff
@@ -353,6 +354,7 @@ def get_supported(
 
         # Current version, current API (built specifically for our Python):
         for abi in abis:
+            assert abi
```

Comment:
Does it work to instead annotate `abi3s` above with `Set[str]` (or similar)? Then you can rely on the type checker without having to do asserts.

Reply:
This file's gonna go away soon enough, when we switch to packaging.tags.

Thus, I'm not too concerned about this.

---

The issue here is that there's an abi attribute in the parameters of the function which is optional string and that name is being reused.

Reply:
Okay, then I would add a code comment that the assert is needed to satisfy the type checker. Otherwise, it’s not clear to someone reading the code why it’s there.

Reply:
Sounds fair to me. I'll add one.

Reply:
> Does it work to instead annotate `abi3s` above with `Set[str]` (or similar)?

AH! This works. Please don't mind me. 🙈